action.yml

```diff
@@ -5,21 +5,21 @@ inputs:
     description: 'Which directory to upload'
     required: true
     default: 'exported-artifacts'
-  distro:
-    description: 'The distribution of artifacts'
-    required: true
-    default: 'el8stream'
 runs:
   using: "composite"
   steps:
     - name: Install createrepo_c if needed
       shell: bash
       run: which createrepo_c || dnf install -y createrepo_c
-    - name: Create DNF repository
+    - name: Create DNF repository if needed
+      shell: bash
+      run: test -d ${{ inputs.directory }}/repodata || createrepo_c ${{ inputs.directory }}
+    - name: Get distro suffix
+      id: eval
       shell: bash
-      run: createrepo_c ${{ inputs.directory }}
+      run: echo "::set-output name=distro::$(rpm --eval %{?dist})"
```

Comment:
Pure magic... do you have any docs on this?

Reply:
Yep it's magic, https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter